presto-main/src/main/java/io/prestosql/sql/analyzer/Analysis.java

```diff
@@ -81,6 +82,7 @@
     @Nullable
     private final Statement root;
     private final List<Expression> parameters;
+    private final Map<NodeRef<Expression>, Integer> parameterPositionMap;
```

Comment:
Make this a `Map<NodeRef<Parameter>, Integer>` and rename the variable to `parameterOrdinals`